cdap-spark-core/src/main/java/co/cask/cdap/app/runtime/spark/SparkRuntimeContextConfig.java

```diff
@@ -139,7 +139,6 @@ public WorkflowProgramInfo getWorkflowProgramInfo() {
       return null;
     }
     WorkflowProgramInfo workflowProgramInfo = GSON.fromJson(info, WorkflowProgramInfo.class);
-    workflowProgramInfo.getWorkflowToken().disablePut();
```

Comment:
This change is necessary. When the driver is running in the Spark AM, the workflow token is deserialized through this method. The driver can modify the token, hence we cannot disable put on the token.